bundler/lib/bundler/definition.rb

```diff
@@ -339,7 +339,10 @@ def lock(file, preserve_unknown_sections = false)
 
       preserve_unknown_sections ||= !updating_major && (Bundler.frozen_bundle? || !(unlocking? || @unlocking_bundler))
 
-      return if file && File.exist?(file) && lockfiles_equal?(@lockfile_contents, contents, preserve_unknown_sections)
+      if file && File.exist?(file) && lockfiles_equal?(@lockfile_contents, contents, preserve_unknown_sections)
+        FileUtils.touch(file)
```

Comment:
Should I wrap this in a `SharedHelpers.filesystem_access` block as well? 🤔

Reply:
Yes, that helper unifies error rescuing for filesystem access problems. Touch is a write to the filesystem.

Reply:
Tell me what you think. Should we copy the frozen bundle check to guard the touch? I know for `lock` it temporarily unfreezes because it's nonsensical otherwise, but this code path will run during install and update, I believe. We should not otherwise change the lockfile if frozen, right?

Reply:
You're referring to the little "Cannot write a changed lockfile while frozen" guard directly below, right? Yes, I also caught that and it seemed sensible not to touch the lockfile when frozen, but I did not add an explicit test about that.

Reply:
Alright, I applied the wrapper. Anything else?